ios/ExpoVideoMetadataModule.swift

```diff
@@ -63,6 +64,11 @@ public class ExpoVideoMetadataModule: Module {
       } else {
         orientation = (transform.a == 1.0) ? "LandscapeRight" : "LandscapeLeft"
       }
+
+      // HDR
+      if #available(iOS 14.0, *) {
+        isHDR = videoTrack.hasMediaCharacteristic(.containsHDRVideo)
+      }
```

Comment:
```suggestion
      // HDR check
      if #available(iOS 14.0, *) {
        if videoTrack.hasMediaCharacteristic(.containsHDRVideo) {
          isHDR = true
        }
      }
```

Untested; please test it.

Another if-condition is needed because we're inside a loop, and the next track might be false, which would override the previous match.

Reply:
I haven't tested it, but found some code that looks pretty similar to the checks that you inserted for android.

```swift
        // HDR check
        if #available(iOS 14.0, *) {
            if videoTrack.hasMediaCharacteristic(.containsHDRVideo) {
                isHDR = true  // Directly set isHDR; do not break to ensure all tracks are processed
            }
        } else {
            // Fallback for earlier iOS versions, check extended color properties to infer HDR
            if let formatDescription = videoTrack.formatDescriptions.first as? CMFormatDescription {
                let extensions = CMFormatDescriptionGetExtensions(formatDescription) as NSDictionary?
                if let colorPrimaries = extensions?["CVImageBufferColorPrimariesKey"] as? String,
                   let transferFunction = extensions?["CVImageBufferTransferFunctionKey"] as? String,
                   let yCbCrMatrix = extensions?["CVImageBufferYCbCrMatrixKey"] as? String {
                    if colorPrimaries == "ITU_R_2020" && 
                       (transferFunction == "SMPTE_ST_2084" || transferFunction == "ITU_R_2100_HLG") && 
                       yCbCrMatrix == "ITU_R_2020" {
                        isHDR = true  // Set isHDR based on color properties; do not break
                    }
                }
            }
        }
```

If you like give it a shot :)

Reply:
> Another if-condition is needed because we're inside a loop

Which loop do you mean? I'm not a "Swifty" but for me it looks like we are only picking the first track?
`if let videoTrack = asset.tracks(withMediaType: .video).first {`

Reply:
Regarding the code < iOS 13 - it sounds promising, but I cannot test it easily. My Xcode does not allow me easily to install iOS 13 simulators. However, if we have a look how little iOS 13 is used out there in the wild right now, I think its fair to just go with iOS 14 and above.

Reply:
Expo SDK52 will drop iOS 13 support, and considering that iOS 18 is around the corner, I decided not to add specific iOS 13 support. And you're right about the loop, not sure why I assumed it